src/bonding-curves/GDACurve.sol

```diff
@@ -56,90 +49,65 @@ contract GDACurve is ICurve, CurveErrorCodes {
         external
         view
         override
-        returns (
-            Error error,
-            uint128 newSpotPrice,
-            uint128 newDelta,
-            uint256 inputValue,
-            uint256 protocolFee
-        )
+        returns (Error error, uint128 newSpotPrice, uint128 newDelta, uint256 inputValue, uint256 protocolFee)
     {
         // NOTE: we assume alpha is > 1, as checked by validateDelta()
         // We only calculate changes for buying 1 or more NFTs
         if (numItems == 0) {
             return (Error.INVALID_NUMITEMS, 0, 0, 0, 0);
         }
 
-        (uint256 alpha, , ) = _parseDelta(delta);
+        uint256 spotPrice_ = uint256(spotPrice);
+        (uint256 alpha,,) = _parseDelta(delta);
+        uint256 alphaPowN = uint256(alpha).powu(numItems);
 
-        uint256 alphaPowN = uint256(alpha).fpow(
-            numItems,
-            FixedPointMathLib.WAD
-        );
+        // TODO: should we cap the time elapsed or decay factor?
```

Comment:
cc @sudo-owen

Reply:
We could cap the max decay factor to 1/100 perhaps? 

Would be enough to capture the range for most use cases and avoids issues with the price getting too low.